sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NormalizeFloatingNumbers.scala

```diff
@@ -144,6 +142,14 @@ object NormalizeFloatingNumbers extends Rule[LogicalPlan] {
       val function = normalize(lv)
       KnownFloatingPointNormalized(ArrayTransform(expr, LambdaFunction(function, Seq(lv))))
 
+    case _ if expr.dataType.isInstanceOf[MapType] =>
+      val MapType(kt, vt, containsNull) = expr.dataType
+      val keys = NamedLambdaVariable("arg", kt, containsNull)
+      val values = NamedLambdaVariable("arg", vt, containsNull)
+      val function = normalize(values)
```

Comment:
why don't we normalize map keys?

Reply:
Do we need to normalize keys? Would a conflict arise for a map that looks like this: `Map(0.0 -> 1, -0.0 -> 2)` as we don't support duplicate keys and `-0.0` would be reduced to `0.0`.

Reply:
You are right, this should be handled when creating the map. We should fix `ArrayBasedMapBuilder` to handle floating points well.

Reply:
Created a [separate PR](https://github.com/apache/spark/pull/45721) for map normalization.